dot/network/finality_proof.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2019 ChainSafe Systems (ON) Corp.
+// This file is part of gossamer.
+//
+// The gossamer library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The gossamer library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the gossamer library. If not, see <http://www.gnu.org/licenses/>.
+
+package network
+
+import (
+	"fmt"
+
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/lib/scale"
+)
+
+// FinalityProofRequest ...
+type FinalityProofRequest struct {
+	Hash    common.Hash
+	Request []byte
+}
+
+// SubProtocol returns the finality-proof sub-protocol.
+func (f *FinalityProofRequest) SubProtocol() string {
+	return finalityProofID
+}
+
+// Decode the message into a FinalityProofRequest.
+func (f *FinalityProofRequest) Decode(b []byte) error {
```

Comment:
I think encode and decode for these types use protobuf, but I just took a look in substrate and couldn't find the proto files. I asked the spec team, will update you on this

Reply:
I will let this scale decoding sit until we have more information from the spec team about this protocol.